src/psyclone/tests/psyGen_test.py

```diff
@@ -2668,17 +2692,609 @@ def test_binaryoperation_can_be_printed():
+    assert fake_parent.symbol_table.lookup("l1").shape == []
+    assert fake_parent.symbol_table.lookup("l1").scope == 'local'
+    assert fake_parent.symbol_table.lookup("l1").is_input is False
+    assert fake_parent.symbol_table.lookup("l1").is_output is False
+
+    reader = FortranStringReader("Real      ::      l2")
+    fparser2spec = Specification_Part(reader).content[0]
+    processor.process_declarations(fake_parent, [fparser2spec], [])
+    assert fake_parent.symbol_table.lookup("l2").name == "l2"
+    assert fake_parent.symbol_table.lookup("l2").datatype == 'real'
+
+    # Test with unsupported data type
+    reader = FortranStringReader("logical      ::      c2")
+    fparser2spec = Specification_Part(reader).content[0]
+    with pytest.raises(NotImplementedError) as error:
+        processor.process_declarations(fake_parent, [fparser2spec], [])
+    assert "Could not process " in str(error.value)
+    assert (". Only 'real', 'integer' and 'character' intrinsic types are"
+            " supported.") in str(error.value)
+
+    # Test with unsupported attribute
+    reader = FortranStringReader("real, public :: p2")
+    fparser2spec = Specification_Part(reader).content[0]
+    with pytest.raises(NotImplementedError) as error:
+        processor.process_declarations(fake_parent, [fparser2spec], [])
+    assert "Could not process " in str(error.value)
+    assert "Unrecognized attribute type " in str(error.value)
+
+    # RHS array specifications are not supported
+    reader = FortranStringReader("integer :: l1(4)")
+    fparser2spec = Specification_Part(reader).content[0]
+    with pytest.raises(NotImplementedError) as error:
+        processor.process_declarations(fake_parent, [fparser2spec], [])
+    assert ("Array specifications after the variable name are not "
+            "supported.") in str(error.value)
+
+    # Initialisations are not supported
+    reader = FortranStringReader("integer :: l1 = 1")
+    fparser2spec = Specification_Part(reader).content[0]
+    with pytest.raises(NotImplementedError) as error:
+        processor.process_declarations(fake_parent, [fparser2spec], [])
+    assert ("Initializations on the declaration statements are not "
+            "supported.") in str(error.value)
+
+    # Char lengths are not supported
+    # TODO: It would be simpler to do just a Specification_Part(reader) instead
+    # of parsing a full program, but fparser/169 needs to be fixed first.
+    reader = FortranStringReader("program dummy\ncharacter :: l*4"
+                                 "\nend program")
+    program = f2008_parser(reader)
+    fparser2spec = program.content[0].content[1].content[0]
+    with pytest.raises(NotImplementedError) as error:
+        processor.process_declarations(fake_parent, [fparser2spec], [])
+    assert ("Character length specifications are not "
+            "supported.") in str(error.value)
+
+
+def test_fparser2astprocessor_process_not_supported_declarations(f2008_parser):
```

Comment:
This test routine does not use `f2008_parser`

Reply:
But they still need the ParserFactory to create a parser right? If I call the test in isolation without the fixture it fails.

Reply:
Oops, yes, you are quite right!